llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -592,27 +718,311 @@ static Error applyAddends(MCCASReader &Reader,
+  uint64_t PrevOffset = CUOffset;
+  uint64_t AbbrevTag = DWARFExtractor.getULEB128(&CUOffset);
+  std::memcpy(&SectionContents[SectionOffset], &AbbrevTag,
+              CUOffset - PrevOffset);
+  SectionOffset += CUOffset - PrevOffset;
+
+  // Return if the abbrev tag is 0, this indicates the end of a sequence of
+  // children DWARFDies
+  if (AbbrevTag == 0)
+    return Error::success();
+
+  // Create a dummy DWARFDie to extract the attributes.
+  DWARFDebugInfoEntry DDIE;
+  auto *AbbrevDecl =
+      DCU.getAbbreviations()->getAbbreviationDeclaration(AbbrevTag);
+  DDIE.setAbbrevDecl(AbbrevDecl);
+  DWARFDie CUDie(&DCU, &DDIE);
+
+  for (unsigned I = 0; I < AbbrevDecl->getNumAttributes(); I++) {
+    auto Form = AbbrevDecl->getFormByIndex(I);
+    auto *U = CUDie.getDwarfUnit();
+    dwarf::FormParams FP = U->getFormParams();
+    auto FormSize = getFormSize(Form, FP, CUData, CUOffset,
+                                DCU.isLittleEndian(), DCU.getAddressByteSize());
+
+    if (!FormSize)
+      consumeError(FormSize.takeError());
+
+    if (Form == llvm::dwarf::DW_FORM_strp ||
+        Form == llvm::dwarf::DW_FORM_sec_offset) {
+      if (*FormSize + FixupOffset > FixupsDataArrayRef.size())
+        return createStringError(
+            inconvertibleErrorCode(),
+            "Trying to read out of bounds from FixupsDataArrayRef");
+      memcpy(&SectionContents[SectionOffset], &FixupsDataArrayRef[FixupOffset],
+             *FormSize);
+      FixupOffset += *FormSize;
+    } else {
+      if (*FormSize + CUOffset > CUData.size())
+        return createStringError(inconvertibleErrorCode(),
+                                 "Trying to read out of bounds from CUData");
+      memcpy(&SectionContents[SectionOffset], &CUData.data()[CUOffset],
+             *FormSize);
+      CUOffset += *FormSize;
+    }
+    SectionOffset += *FormSize;
+  }
+  return Error::success();
+}
+
+Expected<uint64_t>
+DebugInfoSectionRef::materialize(MCCASReader &Reader,
+                                 ArrayRef<char> AbbrevSectionContents,
+                                 raw_ostream *Stream) const {
   // Start a new section for relocations.
   Reader.Relocations.emplace_back();
   auto LoadedSection = LoadedDebugInfoSection::load(*this);
   if (!LoadedSection)
     return LoadedSection.takeError();
 
-  SmallVector<char, 512> SectionContents;
+  SmallVector<char, 0> SectionContents;
   raw_svector_ostream SectionStream(SectionContents);
   unsigned Size = 0;
+  SmallVector<SmallVector<char>> CUContents;
   for (auto [AbbrevOffset, CUData] : LoadedSection->getOffsetAndCUDataRange()) {
     // Copy the data so that we can modify the abbrev offset prior to printing.
     // FIXME: do this with a zero-copy strategy.
     auto MutableCUData = to_vector(CUData);
     if (auto E = getAndSetDebugAbbrevOffsetAndSkip(
-            MutableCUData, Reader.getEndian(), AbbrevOffset);
+            MutableCUData, Reader.getEndian(), AbbrevOffset, false);
```

Comment:
`/* what is this */ false`

See bullet 4 here: https://llvm.org/docs/CodingStandards.html#comment-formatting